File: frontend/src/app/core/auth/service/user.service.ts
import { Injectable } from "@angular/core";
import { environment } from "../../../../environments/environment.dev";
import { Observable } from "rxjs";
import { CreateUserDto, UpdateUserDto, UserDataDto } from "../models/User";
import { HttpClient } from "@angular/common/http";

@Injectable({
    providedIn: "root"
})
export class UserService {
    private apiUrl = `${environment.apiUrl}/users`;

    constructor(private http: HttpClient) {}

    getUserById(id: number): Observable<UserDataDto> {
        return this.http.get<UserDataDto>(`${this.apiUrl}/${id}`);
    }

    getUserByCognitoUserId(cognitoUserId: string): Observable<UserDataDto> {
        return this.http.get<UserDataDto>(`${this.apiUrl}/cognito-id/${cognitoUserId}`);
    }

    createUser(userDto: CreateUserDto): Observable<UserDataDto> {
        return this.http.post<UserDataDto>(this.apiUrl, userDto);
    }

    updateUser(userDto: UpdateUserDto): Observable<UserDataDto> {
        return this.http.put<UserDataDto>(this.apiUrl, userDto);
    }

    deleteUser(id: number): Observable<void> {
        return this.http.delete<void>(`${this.apiUrl}/${id}`);
    }

}

File: backend/src/main/java/com/moviemator/core/user/service/UserService.java
package com.moviemator.core.user.service;

import com.moviemator.core.user.dto.*;
import com.moviemator.shared.search.models.PaginatedResults;
import com.moviemator.shared.search.models.SearchParams;

public interface UserService {

    UserDataDto getUserById(Long id);
    UserDataDto getUserByCognitoUserId(String cognitoUserId);
    PaginatedResults<UserSearchDto> getPublicUsers(SearchParams searchParams);
    PublicUserDataDto getPublicUserById(Long id);
    UserDataDto createUser(CreateUserDto userDto);
    UserDataDto updateUser(UpdateUserDto userDto);
    void deleteUser(Long id);
}
